lib/hdpublickey.js

```diff
@@ -65,6 +65,22 @@ function HDPublicKey(arg) {
   }
 }
 
+
+/**
+ * Verifies that a given path is valid.
+ *
+ * @param {string|number} arg
+ * @return {boolean}
+ */
+HDPublicKey.prototype.isValidPath = function(arg) {
```

Comment:
I thought about that, the issue is being as flexible as the `derive` method (that accepts number/boolean or strings). However another Pro of avoiding the try/catch is we can expose the method as a static function.

I didn't worry too much about the calculation because the class uses a cache, and if you check for a path validity you most probably want to derive it.

Do you think the style improvement worth the change?